index.js

```diff
@@ -6,16 +6,24 @@ module.exports = function(schemas, options) {
   options = options || {};
   return function*(next) {
     try {
+      const request = this.request;
       for (let key in schemas) {
-        this.request[key] = yield (function() {
+        const result = yield (function() {
           return function(cb) {
-            joi.validate(this.request[key], schemas[key], options, cb);
+            joi.validate(request[key], schemas[key], options, cb);
           };
         })();
+        request[key] = result;
+        if (key === 'query') {
+          request._querycache[request.querystring] = result;
+        }
       }
       yield next;
     } catch (err) {
-      err.status = 400;
+      /* istanbul ignore else */
```

Comment:
The tests to actually cover this path were more contrived than I felt was justified..